test/integration/test_connection_management.py

```diff
@@ -95,3 +95,39 @@ def test_http_h2_connection_management_single_request_per_conn_1(http_test_serve
 @pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
 def test_http_h2_connection_management_single_request_per_conn_1(http_test_server_fixture):
   connection_management_test_request_per_connection(http_test_server_fixture, 5, True)
+
+
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_h1_pool_strategy(http_test_server_fixture):
+  """
+  Test that with the "mru" strategy only the first created connection gets to send requests.
+  Then, with the "lru" strategy, we expect the other connection to be used as well.
+  """
+
+  def countLogLinesWithSubstring(logs, substring):
+    return len([line for line in logs.split(os.linesep) if substring in line])
+
+  _, logs = http_test_server_fixture.runNighthawkClient([
+      "--rps 20", "-v", "trace", "--connections", "2", "--prefetch-connections",
+      "--experimental-h1-connection-reuse-strategy", "mru", "--termination-predicate",
+      "benchmark.http_2xx:10",
+      http_test_server_fixture.getTestServerRootUri()
+  ])
+
+  requests = 60
+  connections = 3
+  assertNotIn("[C1] message complete", logs)
```

Comment:
It would be nice if Envoy has or could be taught to have better direct stats here. Relying on log messages is somewhat fragile.

Reply:
I agree, what's here today is best-effort based on the means currently available, but going forward perhaps a histogram (or just an average) which tracks replies/connection could be useful to get a sense of connection re-use?

Reply:
Something like that, sure. Just wanted to make sure it was on your radar. Thanks.